src/middleware/auth/logto-helper.ts

```diff
@@ -287,58 +258,104 @@ export class LogToHelper extends OAuthProvider implements IOAuthProvider {
 		}
 	}
 
-	private async getRoleInfo(roleId: string): Promise<ICommonErrorResponse> {
-		const uri = new URL(`/api/roles/${roleId}`, process.env.LOGTO_ENDPOINT);
-		try {
-			return await this.getToLogto(uri, 'GET');
-		} catch (err) {
-			return this.returnError(StatusCodes.BAD_GATEWAY, `getRoleInfo ${err}`);
-		}
-	}
-
-	private async assignDefaultRoleForUser(userId: string, roleId: string): Promise<ICommonErrorResponse> {
-		const userInfo = await this.getUserInfo(userId);
-		const uri = new URL(`/api/users/${userId}/roles`, process.env.LOGTO_ENDPOINT);
+	async assignCustomerPlanRoles(userId: string, planType: SupportedPlanTypes): Promise<ICommonErrorResponse> {
+		const [userInfo, userRoles] = await Promise.all([this.getUserInfo(userId), this.getRolesForUser(userId)]);
+		console.log('assignCustomerPlanRoles: userInfo: ', userInfo);
+		console.log('assignCustomerPlanRoles: userRoles: ', userRoles);
 
 		if (userInfo.status !== StatusCodes.OK) {
-			return this.returnError(
-				StatusCodes.BAD_GATEWAY,
-				`Could not fetch the info about role with roleId ${roleId}`
-			);
+			return this.returnError(StatusCodes.BAD_GATEWAY, 'Could not fetch the info about the user');
 		}
 		// Means that user exists
-		if (userInfo.data.isSuspended === 'true') {
+		if (userInfo.data.isSuspended) {
 			return this.returnError(StatusCodes.FORBIDDEN, 'User is suspended');
 		}
-		// Means it's not suspended
-		const role = await this.getRoleInfo(roleId);
-		if (role.status !== StatusCodes.OK) {
-			return this.returnError(
-				StatusCodes.BAD_GATEWAY,
-				`Could not fetch the info about user with userId ${userId} because of error from authority server: ${role.error}`
-			);
+
+		if (userRoles.status !== StatusCodes.OK) {
+			return this.returnError(StatusCodes.BAD_GATEWAY, 'Could not fetch the info about user roles');
```

Comment:
Same logic applies here regarding the status code returned.

Reply:
I've switched to underlying status code and changed the response code in the request sender wrapper to use `503 Service Unavailable` when we actually can't reach it